include/lsst/afw/table/types.h

```diff
@@ -25,20 +25,20 @@
  */
 
 // Scalar types: those that can serve as elements for other types, and use the default FieldBase template.
-#define AFW_TABLE_SCALAR_FIELD_TYPE_N 6
-#define AFW_TABLE_SCALAR_FIELD_TYPES RecordId, std::uint16_t, std::int32_t, float, double, Angle
+#define AFW_TABLE_SCALAR_FIELD_TYPE_N 7
+#define AFW_TABLE_SCALAR_FIELD_TYPES RecordId, std::uint16_t, std::int32_t, float, double, Angle, std::uint8_t
 #define AFW_TABLE_SCALAR_FIELD_TYPE_TUPLE BOOST_PP_LPAREN() AFW_TABLE_SCALAR_FIELD_TYPES BOOST_PP_RPAREN()
 
 // Arrays types: the types we allow for Array fields.
-#define AFW_TABLE_ARRAY_FIELD_TYPE_N 4
-#define AFW_TABLE_ARRAY_FIELD_TYPES std::uint16_t, int, float, double
+#define AFW_TABLE_ARRAY_FIELD_TYPE_N 5
+#define AFW_TABLE_ARRAY_FIELD_TYPES std::uint16_t, int, float, double, std::uint8_t
 #define AFW_TABLE_ARRAY_FIELD_TYPE_TUPLE BOOST_PP_LPAREN() AFW_TABLE_ARRAY_FIELD_TYPES BOOST_PP_RPAREN()
 
 // Field types: all the types we allow for fields.
-#define AFW_TABLE_FIELD_TYPE_N 12
+#define AFW_TABLE_FIELD_TYPE_N 14
 #define AFW_TABLE_FIELD_TYPES                                                                        \
     AFW_TABLE_SCALAR_FIELD_TYPES, Flag, std::string, Array<std::uint16_t>, Array<int>, Array<float>, \
```

Comment:
Unrelated, but is there any reason why we use `Array<int>` while the scalar is specifically qualified as `std::int32_t`?

Reply:
I have no idea. @TallJimbo?

Reply:
`std::int32_t` would be better, but changing it would require concerted changes across a lot of the codebase.  Should probably be spun off into another ticket.